packages/snaps-execution-environments/src/common/validation.ts

```diff
@@ -207,3 +207,26 @@ export type ExecuteSnap = RequestFunction<
   OkResponse
 >;
 export type SnapRpc = RequestFunction<SnapRpcRequestArguments, SnapRpcResponse>;
+
+/**
+ * Validate and sanitize JSON structure.
+ *
+ * Note:
+ * This function will stringify and then parse the object provided
+ * to ensure that there are no getters which can have side effects
+ * that can cause security issues.
+ *
+ * @param value - JSON structure to be processed.
+ * @returns Sanitized JSON structure.
+ */
+export function sanitizeJsonStructure(value: unknown): Json {
+  try {
+    assertStruct(value, JsonStruct);
+    const sanitizedJsonStruct = JSON.parse(JSON.stringify(value));
+    assertStruct(sanitizedJsonStruct, JsonStruct);
```

Comment:
Can this not be handled in `utils` with a coercion?

Reply:
Not sure I understand. Essentially what we're doing here is a coercion. What would the difference between this and a Superstruct coercion be?

Reply:
Mostly intent. Currently the validation functions were assumed to be safe and were misused due to not using the coercion result.

The idea of this PR is to consider any output of a snap to be malicious and to make it safe, which is different from checking if a structure of data is valid.

This ensures that the readers understand that before sanitization function is called, it's explicitly unsafe, quarantine of sorts.

In the future, there might be more sanitisation required, which doesn't necessarily need to go into validation layer as we use validation in multiple other places, where there's no possibility of malicious data, just mis-structured data.

Reply:
> The idea of this PR is to consider any output of a snap to be malicious and to make it safe, which is different from checking if a structure of data is valid.

I understand that, but I'm not sure if we should be doing this here, or in `utils` (using a Superstruct coercion).

> In the future, there might be more sanitisation required, which doesn't necessarily need to go into validation layer as we use validation in multiple other places, where there's no possibility of malicious data, just mis-structured data.

This is why the `UnsafeJsonStruct` and `JsonStruct` existed.

The `JsonStruct` can be something like

```ts
const JsonStruct = coerce(UnsafeJsonStruct, UnsafeJsonStruct, (value) => {
  const sanitizedValue = JSON.parse(JSON.stringify(value));
  assertStruct(sanitizedValue, UnsafeJsonStruct);

  return sanitizedValue;
});
```

Which can then be used like this:

```ts
const value = create(json, JsonStruct);
```

Even better if we made a function for it:

```ts
function getValidatedJson<Type>(value: Type) {
  return create(json, JsonStruct);
}
```

I don't see why this logic should be moved to `snaps-monorepo`.

Reply:
We could have `sanitizeJson` and `isJson` both in utils if you prefer.

I'd say the sanitisation is snap unique because it's strongly focused on SES returned objects, but either option works for me

Reply:
Alright, the solution is refactored according to your comments and suggestions. Let me know if this one is better.

Changes on this PR are following https://github.com/MetaMask/utils/pull/103 and once updated with the new package version, CI should be passing.